### PR TITLE
feat(#11244): verbose git flags in aliases

### PR DESCRIPTION
**Pull request by jdevries3133**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

Use verbose flags in git aliases

## Other comments:

I figure these commits will be squashed and merged so I'm not sure if this nails the commit guidelines, but feel free to merge / reword as you see fit.

### CODE REVIEW

**Review by esemeniuc — APPROVED** [review]

**Comment by esemeniuc** [comment]
Would love to see this merged!

**Comment by carlosala** [comment]
Hey! Could you explain a bit why this would be great? Thanks!

**Comment by rwmitchell** [comment]
It makes the aliases readable when expanding an alias to see what it is.

**Comment by carlosala** [comment]
It makes total sense!

File: plugins/git/git.plugin.zsh [hunk 1]
```diff
@@ -103,20 +103,20 @@ function gccd() {
...
+alias gclean='git clean --interactive -d'
 alias gpristine='git reset --hard && git clean -dffx'
 alias gcm='git checkout $(git_main_branch)'
 alias gcd='git checkout $(git_develop_branch)'
-alias gcmsg='git commit -m'
+alias gcmsg='git commit --message'
 alias gco='git checkout'
 alias gcor='git checkout --recurse-submodules'
-alias gcount='git shortlog -sn'
+alias gcount='git shortlog --summary -n'
```
Comment (tjarbo) [inline plugins/git/git.plugin.zsh:113]:
```suggestion
alias gcount='git shortlog --summary --numbered'
```
https://git-scm.com/docs/git-shortlog#Documentation/git-shortlog.txt---numbered

File: plugins/git/git.plugin.zsh [hunk 2]
```diff
@@ -267,15 +267,15 @@ alias grst='git restore --staged'
 alias grt='cd "$(git rev-parse --show-toplevel || echo .)"'
 alias gru='git reset --'
 alias grup='git remote update'
-alias grv='git remote -v'
+alias grv='git remote --verbose'
 
-alias gsb='git status -sb'
+alias gsb='git status --short -b'
```
Comment (tjarbo) [inline plugins/git/git.plugin.zsh:272]:
```suggestion
alias gsb='git status --short --branch'
```
https://git-scm.com/docs/git-status#Documentation/git-status.txt---branch

File: plugins/git/git.plugin.zsh (hunk 1 above)
Comment (jdevries3133) [reply plugins/git/git.plugin.zsh:113]: Oops, there's no reason that was just a mistake! I used some regexes to fix most flags, then came back for the grouped ones (i.e. `-sn`) by hand. Good catch! I'll fix these.

**Comment by jdevries3133** [comment]
OK, I took another look and fixed the two that @tjarbo pointed out as well as a few more.

**Comment by nicolasbarra** [comment]
Excited to see this merged after the conflicts have been resolved!

File: plugins/git/git.plugin.zsh [hunk 3]
```diff
@@ -103,20 +103,20 @@ function gccd() {
...
-alias gcount='git shortlog -sn'
+alias gcount='git shortlog --summary --numbered'
 alias gcp='git cherry-pick'
 alias gcpa='git cherry-pick --abort'
 alias gcpc='git cherry-pick --continue'
 alias gcs='git commit -S'
-alias gcss='git commit -S -s'
-alias gcssm='git commit -S -s -m'
+alias gcss='git commit -S --signoff'
+alias gcssm='git commit -S --signoff --message'
```
Comment (tjarbo) [inline plugins/git/git.plugin.zsh:119]: `-S` should be replaceable with `--gpg-sign` according to the [git documentation](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--Sltkeyidgt)

**Review by tjarbo — CHANGES_REQUESTED** [review]
Hi @jdevries3133,
really great! Just found one missing parameter that is replaceable with a more descriptive one. An execption is the `gunwip` alias, where the `-n` could be replaced as well, but I would say that it does not make that much sense.

best regards
tjarbo

File: plugins/git/git.plugin.zsh (hunk 3 above)
Comment (jdevries3133) [reply plugins/git/git.plugin.zsh]: OK I finally got the chance to fix this up; I am pretty sure there are no stragglers left but looking forward to your review. Thanks for your patience!

**Comment by jdevries3133** [comment]
Also, merge conflicts have been resolved. Two notes while merging with upstream:

- I noticed that `gcn!` was removed; I was sure to carry that change forward
- I noticed that Carlos Sala had a patch (0145d744a9c4c11f00992f7f3ad9555bc8ac6177) to change `gunwip` message; I was careful to include that change as well

**Review by tjarbo — CHANGES_REQUESTED** [review]
Hi @jdevries3133,
I just found a little mistake that sneaked in during your last commits. According to the logs, all your changes for `git commit` (line 85-95) got reverted. Please check again :) 

https://github.com/ohmyzsh/ohmyzsh/pull/11259/commits/7300030e1f8c71cb4910ed3bae9a505436099231#diff-bd72f3b7bb82839b6e8f34b96e7fc5ec3c923d24b79bec12427df62f07136678R85

best regards
tjarbo

**Comment by jdevries3133** [comment]
> According to the logs, all your changes for `git commit` (line 85-95) got reverted.

Ah, nice catch; fixed in 2684f73db1f21c0b48e60b5c6470d479efd4150b

**Review by tjarbo — APPROVED** [review]
Perfect @jdevries3133! Could not find anything else left 👍

**Comment by carlosala** [comment]
This PR is becoming really great! Did you checked that every alias is documented in the README, for example something like #10101? And maybe we could also bundle the idea of #9743 in this PR, what do you think @jdevries3133?

**Comment by jdevries3133** [comment]
> Did you checked that every alias is documented in the README, for example something like #10101?

I'm pretty sure that every alias documentation is updated correctly, that's certainly my intention, but feel free to review the changes of course. 

> And maybe we could also bundle the idea of #9743 in this PR, what do you think @jdevries3133?

I already stupidly made an attempt at formatting the table which caused a bit of a gnarly merge conflict. I want to avoid sweeping changes to the readme, but I'm happy to do this in a follow up PR if we can turn it around quickly to minimize any conflict issues.

**Comment by carlosala** [comment]
Thanks for your contribution @jdevries3133 and @tjarbo for the reviews. I've just squashed and pushed all this PR to 35454190f6aeae9b9f16acdbdd0cf72c32610ee8.
I'm really happy we managed to close this!

**Pull request closed without merging**